packages/core/src/api/blockManipulation/commands/updateBlock/updateBlock.ts

```diff
@@ -127,17 +132,19 @@ export function updateBlockTr<
     // currently, we calculate the new node and replace the entire node with the desired new node.
     // for this, we do a nodeToBlock on the existing block to get the children.
     // it would be cleaner to use a ReplaceAroundStep, but this is a bit simpler and it's quite an edge case
-    const existingBlock = nodeToBlock(blockInfo.bnBlock.node, pmSchema);
+    const newFullBlock = partialBlockToFullBlock(schema, block);
```

Comment:
```suggestion
    const newBlock = partialBlockToFullBlock(schema, block);
```

There are partial blocks & blocks. Full block gives a third name which is not helpful